### PR TITLE
✨ Dynamic volumetric weight

### PR DESCRIPTION
**Pull request by M4tini**
https://myparcelcombv.atlassian.net/browse/MP-4677
---
- Added the volumetric weight divisor to the `Service` resource.
- Removed the volumetric weight divisor from the `ServiceRate` resource.
  - I added this when implementing the bracket price, because the api-spec listed it, but it was never implemented.
- Added the volumetric weight divisor factor to the `Contract` resource.
- Replaced the usage of `filter[volumetric_weight]` with `filter[volume]` when fetching rates.
- Updated the `PriceCalculator` to properly calculate the billable weight and respect the volumetric weight.

### CODE REVIEW

File: src/Shipments/PriceCalculator.php [hunk 1]
```diff
@@ -124,28 +142,29 @@ private function determineServiceRateForShipment(ShipmentInterface $shipment): S
...
 
-        $shipmentOptionIds = array_map(function (ServiceOptionInterface $serviceOption) {
-            return $serviceOption->getId();
-        }, $shipment->getServiceOptions());
+        $shipmentOptionIds = array_map(
+            fn (ServiceOptionInterface $serviceOption) => $serviceOption->getId(),
+            $shipment->getServiceOptions(),
+        );
 
         array_filter($serviceRates, function (ServiceRateInterface $serviceRate) use ($shipmentOptionIds) {
```
Comment (zwaans) [inline src/Shipments/PriceCalculator.php:154]: What exactly is being done here? dont we have to store the outcome of `array_filter` in a new variable?

File: tests/Traits/MocksContract.php [hunk 2]
```diff
@@ -96,6 +96,7 @@ protected function getMockedShipment(
         if ($service) {
             $shipment->method('getService')->willReturn($service);
         }
+        $shipment->method('calculateVolumeInMm3')->willReturn($volumeInMm3);
```
Comment (zwaans) [inline tests/Traits/MocksContract.php:99]: Lets also unit test the `calculateVolumeInMm3` and `calculateVolumeInDm` methods in a unit test?

File: src/Shipments/PriceCalculator.php (hunk 1 above)
Comment (M4tini) [reply src/Shipments/PriceCalculator.php:154]: Ehm, good point, I only refactored part of it, but it seems this code is not doing anything useful.

The purpose is to only return rates that support the already set options (if any) on that shipment. I'm going to fix this

File: src/Resources/Interfaces/ShipmentInterface.php [hunk 3]
```diff
@@ -145,6 +145,10 @@ public function getVolumetricWeight(): ?int;
      */
     public function setVolumetricWeight(?int $volumetricWeight): self;
 
+    public function calculateVolumeInMm3(): ?int;
```
[the remark this answers is not in the document]
Comment (M4tini) [reply src/Resources/Interfaces/ShipmentInterface.php:148]: I was first thinking: we do not need this and a shipment's weight is always `grams` and the physical properties are always `mm`. But then I remembered that our SDK functions do have a second parameter to support different units, so I will make this dynamic as well 👌🏻

**Review by zwaans — APPROVED** [review]

**Pull request merged**